lapack/unit_test/Test_Lapack_svd.hpp

```diff
@@ -0,0 +1,321 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+#include <Kokkos_Core.hpp>
+#include <KokkosKernels_TestUtils.hpp>
+#include <KokkosBlas3_gemm.hpp>
+
+#include <KokkosLapack_svd.hpp>
+
+namespace Test {
+
+template <class AMatrix, class SVector, class UMatrix, class VMatrix>
+void check_triple_product(const AMatrix& A, const SVector& S, const UMatrix& U,
```

Comment:
@lucbv I am not sure about this. Please correct me if I am wrong. Are you using this function later?

Reply:
This should be used for the randomly generated matrices test. One issue is that the diagonal matrix S is not going to work with gemm, I'll think of a solution for that...